tfexec/options.go

```diff
@@ -172,6 +176,39 @@ func PluginDir(pluginDir string) *PluginDirOption {
 	return &PluginDirOption{pluginDir}
 }
 
+type ReattachInfo map[string]ReattachConfig
+
+// ReattachConfig holds the information Terraform needs to be able to attach
+// itself to a provider process, so it can drive the process.
+type ReattachConfig struct {
+	Protocol string
+	Pid      int
+	Test     bool
+	Addr     ReattachConfigAddr
+}
+
+// ReattachConfigAddr is a JSON-encoding friendly version of net.Addr.
+type ReattachConfigAddr struct {
+	Network string
+	String  string
+}
+
+type ReattachOption struct {
+	info ReattachInfo
+}
+
+func marshalReattachString(info ReattachInfo) (string, error) {
```

Comment:
you should just hang this off the type to not add to the global namespace
```suggestion
func (info ReattachInfo) marshalReattachString() (string, error) {
```